contracts/DepositManager.sol

```diff
@@ -11,41 +11,92 @@ import { POB } from "./POB.sol";
 import { Governance } from "./Governance.sol";
 import { Rollup } from "./Rollup.sol";
 
-contract DepositManager {
-    using Types for Types.UserState;
-    MTUtils public merkleUtils;
-    Registry public nameRegistry;
-    bytes32[] public pendingDeposits;
-    address public vault;
-    mapping(uint256 => bytes32) pendingFilledSubtrees;
-    uint256 public firstElement = 1;
-    uint256 public lastElement = 0;
-    uint256 public depositSubTreesPackaged = 0;
-
-    function enqueue(bytes32 newDepositSubtree) public {
-        lastElement += 1;
-        pendingFilledSubtrees[lastElement] = newDepositSubtree;
-        depositSubTreesPackaged++;
+contract SubtreeQueue {
+    // Each element of the queue is a root of a subtree of deposits.
+    mapping(uint256 => bytes32) queue;
+    uint256 public front = 1;
+    uint256 public back = 0;
+
+    function enqueue(bytes32 subtreeRoot) internal {
+        back += 1;
+        queue[back] = subtreeRoot;
+    }
+
+    function dequeue() internal returns (bytes32 subtreeRoot) {
+        require(back >= front, "Deposit Core: Queue should be non-empty");
+        subtreeRoot = queue[front];
+        delete queue[front];
+        front += 1;
     }
+}
+
+contract DepositCore is SubtreeQueue {
+    // An element in this array is a deposit tree root of any depth.
+    // It could be just a leaf of a new deposit or
+    // a root of a full grown subtree.
+    bytes32[] public babyTrees;
+    uint256 public depositCount = 0;
+
+    uint256 public MAX_SUBTREE_SIZE;
```

Comment:
`MAX_SUBTREE_SIZE` should be initialized in a constructor. But if I put a constructor in the DepositCore contract, the typechain couldn't generate a constructor for the DepositManager contract.
It's a minor issue though.